solidity/contracts/token/HypValue.sol

```diff
@@ -0,0 +1,166 @@
+     * @param _owner The owner of the contract
+     */
+    function initialize(
+        address _valuehook,
+        address _interchainSecurityModule,
+        address _owner
+    ) public initializer {
+        _MailboxClient_initialize(
+            _valuehook,
+            _interchainSecurityModule,
+            _owner
+        );
+    }
+
+    // ============ External Functions ============
+
+    /**
+     * @inheritdoc TokenRouter
+     * @dev use _hook with caution, make sure that this hook can handle msg.value transfer using the metadata.msgValue()
+     */
+    function transferRemote(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amount,
+        bytes calldata _hookMetadata,
+        address _hook
+    ) external payable virtual override returns (bytes32 messageId) {
+        uint256 quote = _checkSufficientValue(_destination, _amount, _hook);
+
+        bytes memory hookMetadata = StandardHookMetadata.overrideMsgValue(
+            _hookMetadata,
+            _amount
+        );
+
+        return
+            _transferRemote(
+                _destination,
+                _recipient,
+                _amount,
+                _amount + quote,
+                hookMetadata,
+                _hook
+            );
+    }
+
+    /// @inheritdoc TokenRouter
+    function transferRemote(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amount
+    ) external payable virtual override returns (bytes32 messageId) {
+        uint256 quote = _checkSufficientValue(
+            _destination,
+            _amount,
+            address(hook)
+        );
+        bytes memory hookMetadata = StandardHookMetadata.formatMetadata(
+            _amount,
+            destinationGas[_destination],
+            msg.sender,
+            ""
+        );
+
+        return
+            _transferRemote(
+                _destination,
+                _recipient,
+                _amount,
+                _amount + quote,
+                hookMetadata,
+                address(hook)
+            );
+    }
+
+    // ============ Internal Functions ============
+
+    /**
+     * @inheritdoc TokenRouter
+     * @dev No token metadata is needed for value transfers
+     */
+    function _transferFromSender(
+        uint256
+    ) internal pure override returns (bytes memory) {
+        return bytes(""); // no token metadata
+    }
+
+    /**
+     * @inheritdoc TokenRouter
+     * @dev Sends the value to the recipient
+     */
+    function _transferTo(
+        address _recipient,
+        uint256 _amount,
+        bytes calldata // no token metadata
+    ) internal virtual override {
+        Address.sendValue(payable(_recipient), _amount);
+    }
+
+    /**
+     * @inheritdoc TokenRouter
+     * @dev This contract doesn't hold value
+     */
+    function balanceOf(
+        address /* _account */
+    ) external pure override returns (uint256) {
+        return 0;
+    }
```

Comment:
the `HypNative` treats this differently, another reason to dedupe
```
function balanceOf(
        address _account
    ) external view override returns (uint256) {
        return _account.balance;
    }
```

Reply:
Disagreed, this will be create more confusion. You get value from a different source and now the value is not reflective of what you actually bridged over. I prefer to make it 0 for HypNative too.

Reply:
Another thing, you go bridge to HypNative, and now your wallet will show double the value, one as native value and second as HypNative balance.

Reply:
> Disagreed, this will be create more confusion. You get value from a different source and now the value is not reflective of what you actually bridged over. I prefer to make it 0 for HypNative too.

This is already supported in the UI for HypNative. I am adamant that we should be consistent. Making HypNative 0 would be a breaking change that would brick all existing deployed warp routes.

Reply:
> Another thing, you go bridge to HypNative, and now your wallet will show double the value, one as native value and second as HypNative balance.

Your wallet does not group native balance and token balances together.

Reply:
Disagreed, but don't care too much, changing to account.balance.